cupy/prof/timerange.py

```diff
@@ -0,0 +1,90 @@
+import contextlib
+import functools
+
+from cupy.cuda import nvtx  # NOQA
+
+
+@contextlib.contextmanager
+def timerange(message, id_color=-1):
+    """A context manager to describe the enclosed block as a nested range
+
+    >>> with cupy.prof.timerange('some range in green', 0):
+    ...    # do something you want to measure
+    ...    pass
+
+    Args:
+        message (str): Name of a range.
+        id_color (int): ID of color for a range.
+
+    .. seealso:: :func:`cupy.cuda.nvtx.RangePush`
+        :func:`cupy.cuda.nvtx.RangePop`
+    """
+    nvtx.RangePush(message, id_color)
+    try:
+        yield
+    finally:
+        nvtx.RangePop()
+
+
+@contextlib.contextmanager
+def timerangeC(message, color=0):
```

Comment:
I feel it is confusing a little. How about adding an optional argument `color` to `time_range` to select `color` or `color_id`?